lib/theme_check/tags.rb

```diff
@@ -169,14 +169,32 @@ class Javascript < Liquid::Raw; end
 
     class Stylesheet < Liquid::Raw; end
 
-    Liquid::Template.register_tag('form', Form)
-    Liquid::Template.register_tag('layout', Layout)
-    Liquid::Template.register_tag('render', Render)
-    Liquid::Template.register_tag('paginate', Paginate)
-    Liquid::Template.register_tag('section', Section)
-    Liquid::Template.register_tag('style', Style)
-    Liquid::Template.register_tag('schema', Schema)
-    Liquid::Template.register_tag('javascript', Javascript)
-    Liquid::Template.register_tag('stylesheet', Stylesheet)
+    class << self
+      attr_writer :register_tags
+
+      def register_tags?
+        @register_tags
+      end
+
+      def register_tag(name, klass)
+        return if Liquid::Template.tags[name]
+        Liquid::Template.register_tag(name, klass)
+      end
+
+      def register_tags!
```

Comment:
there were a lot of rebase conflicts around this area. best to give it a close inspection to make sure I've resolved them correctly.